lib/block-supports/layout.php

```diff
@@ -233,29 +233,33 @@ function( $matches ) {
  * to avoid breaking styles relying on that div.
  *
  * @param string $block_content Rendered block content.
+ * @param  array  $block        Block object.
  * @return string Filtered block content.
  */
-function gutenberg_restore_image_outer_container( $block_content ) {
-	$image_with_align = '/(^\s*<figure\b[^>]*)\bwp-block-image\b([^"]*\b(?:alignleft|alignright|aligncenter)\b[^>]*>.*<\/figure>)/U';
+function gutenberg_restore_image_outer_container( $block_content, $block ) {
+	$image_with_align = '/(^\s*<figure.*?class=["\'])(.*\bwp-block-image\b[^"\']*\b(?:alignleft|alignright|aligncenter)\b[^"\']*)(["\'][^>]*>.*<\/figure>)/iU';
```

Comment:
I like that this is all one regex 🙂

Reviewing the regex left to right...

- I'd put a `\b` after `figure` so won't match e.g. `<figureclass="`.
- Better to use `[^]` instead of `.` where possible as then we're explicit about what we expect and won't overmatch.- 
- Because we have the `U` flag enabled, `.*?` is redundant and could just be `.*`.
- Technically there could be whitespace before and after the `=`. Not a big deal though—obviously we can't get all of the HTML spec into a regular expression. 
- Consider making this an extended regex so that it's easier to read.

All minor notes—feel free to ignore. The most important thing is that it works 😀

```suggestion
	$image_with_align = "
/# 1) everything up to the class attribute contents
(
	^\s*
	<figure\b
	[^>]*
	\bclass=
	[\"']
)
# 2) the class attribute contents
(
	[^\"']*
	\bwp-block-image\b
	[^\"']*
	\b(?:alignleft|alignright|aligncenter)\b
	[^\"']*
)
# 3) everything after the class attribute contents
(
	[\"']
	[^>]*
	>
	.*
	<\/figure>
)/iUx
	";
```